packages/react-ui/src/features/pieces/lib/pieces-hook.ts

```diff
@@ -105,3 +67,73 @@ export const piecesHooks = {
     });
   },
 };
+
+function stepMetadataQueryBuilder({ step }: { step: Action | Trigger }) {
+  const isPieceStep =
+    step.type === ActionType.PIECE || step.type === TriggerType.PIECE;
+  const pieceName = isPieceStep ? step.settings.pieceName : `${step.type}`;
+  const pieceVersion = isPieceStep ? step.settings.pieceVersion : ``;
```

Comment:
This is super hacky, just make the getStepMetadata takes step and It would use pieceName and pieceVerison when it's really required.